src/Protocol/Diaspora.php

```diff
@@ -942,31 +942,41 @@ private static function key($handle)
 	 * @brief Fetches data for a given handle
 	 *
 	 * @param string $handle The handle
+	 * @param boolean $update Update the profile
 	 *
 	 * @return array the queried data
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 * @throws \ImagickException
 	 */
-	public static function personByHandle($handle)
+	public static function personByHandle($handle, $update = null)
```

Comment:
Please change the default value for `$update` to `false` since it's a boolean and there's no check for `null` in the function body.

Reply:
There is a difference between ```false``` and ```null``` in this case. ```false``` means: "Never update", ```null``` means "update once in a while".

Reply:
Boolean values should only be dual .. true, false .. a "third" state is a code smell :-/ .. maybe you could use a int with descriptive constants..

Reply:
I concur with @nupplaphil , I didn't see the null check but the intended usage doesn't fit a boolean.

Reply:
Well variables can be ```null``` - so I don't see why why we shouldn't get use of this. In fact I would had to rebuild several places in the code for changing this as ```APContact::getByURL``` is doing so since the beginning (and of course it is documented in that function documentation.

Using boolean values do always add the need to look at the documentation to see what these values stand for.

And: At many, many places in the code we are using boolean variables to trigger the update or the use of the cache. (In fact I would like I would like to change all these functions so that a ```true``` always triggers an update instead of triggering the cache in some parts of the code)

Reply:
The parameter documentation doesn't mention null value as being its own thing at least in `Diaspora::personByHandle`. An int flag field would be more explicit in that regard.

Reply:
@MrPetovan now it does.